src/server/pfs/server/commit_store.go

```diff
@@ -216,58 +212,52 @@ func dropTotal(tx *pachsql.Tx, tr track.Tracker, commit *pfs.Commit) error {
 	}
 	trackID := commitTotalTrackerID(commit, *id)
 	if err := tr.DeleteTx(tx, trackID); err != nil {
-		return errors.EnsureStack(err)
+		return errors.Wrap(err, "drop total")
 	}
-	_, err = tx.Exec(`DELETE FROM pfs.commit_totals WHERE commit_id = $1`, pfsdb.CommitKey(commit))
-	return errors.EnsureStack(err)
+	_, err = tx.Exec(`UPDATE pfs.commits SET total_fileset_id = NULL WHERE int_id = $1`, commit.ID)
+	return errors.Wrap(err, "drop total")
 }
 
-func setTotal(tx *pachsql.Tx, tr track.Tracker, commit *pfs.Commit, id fileset.ID) error {
+func setTotal(tx *pachsql.Tx, tr track.Tracker, commit *pfsdb.CommitWithID, id fileset.ID) error {
 	oid := commitTotalTrackerID(commit, id)
 	pointsTo := []string{id.TrackerID()}
 	if err := tr.CreateTx(tx, oid, pointsTo, track.NoTTL); err != nil {
-		return errors.EnsureStack(err)
+		return errors.Wrap(err, "set total")
 	}
-	if err := checkCommitDB(tx, commit); err != nil {
-		return errors.EnsureStack(err)
-	}
-	_, err := tx.Exec(`INSERT INTO pfs.commit_totals (commit_id, fileset_id)
-	VALUES ($1, $2)
-	ON CONFLICT (commit_id) DO UPDATE
-	SET fileset_id = $2
-	WHERE commit_totals.commit_id = $1
-	`, pfsdb.CommitKey(commit), id)
-	return errors.EnsureStack(err)
+	_, err := tx.Exec(`UPDATE pfs.commits SET total_fileset_id = $1 WHERE int_id = $2`, id, commit.ID)
+	return errors.Wrap(err, "set total")
 }
 
-func setDiff(tx *pachsql.Tx, tr track.Tracker, commit *pfs.Commit, id fileset.ID) error {
+func setDiff(tx *pachsql.Tx, tr track.Tracker, commit *pfsdb.CommitWithID, id fileset.ID) error {
 	oid := commitDiffTrackerID(commit, id)
 	pointsTo := []string{id.TrackerID()}
 	if err := tr.CreateTx(tx, oid, pointsTo, track.NoTTL); err != nil {
-		return errors.EnsureStack(err)
+		return errors.Wrap(err, "set diff")
 	}
 	if err := checkCommitDB(tx, commit); err != nil {
-		return errors.EnsureStack(err)
+		return errors.Wrap(err, "set diff")
 	}
 	_, err := tx.Exec(`INSERT INTO pfs.commit_diffs (commit_id, fileset_id)
 	VALUES ($1, $2)
-	`, pfsdb.CommitKey(commit), id)
-	return errors.EnsureStack(err)
+	`, commit.Commit.Key(), id)
+	return errors.Wrap(err, "set diff")
 }
 
-func checkCommitDB(tx *pachsql.Tx, commit *pfs.Commit) error {
-	var id *int = new(int)
-	err := tx.QueryRow(`SELECT int_id FROM pfs.commits AS c
-		WHERE c.commit_id = $1`,
-		pfsdb.CommitKey(commit),
-	).Scan(id)
-	return errors.EnsureStack(err)
+func checkCommitDB(tx *pachsql.Tx, commit *pfsdb.CommitWithID) error {
```

Comment:
I think we should just require that ID is set. This seems like a useful global invariant for `pfsdb.CommitWithID`, but we can check it in the commit store just to play it safe in the short term.

Reply:
Right

Reply:
I think we should just remove this function, and add a check for id being set where it is called.